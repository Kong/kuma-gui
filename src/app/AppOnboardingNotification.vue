<template>
  <div
    v-if="alertClosed === false"
    class="onboarding-check"
  >
    <KAlert
      appearance="success"
      dismiss-type="icon"
      @closed="closeAlert"
    >
      <template #alertMessage>
        <div class="alert-content">
          <div>
            <strong>Welcome to {{ t('common.product.name') }}!</strong> We've detected that you don't have any data plane proxies running yet. We've created an onboarding process to help you!
          </div>

          <div>
            <KButton
              appearance="primary"
              size="small"
              class="action-button"
              :to="{ name: 'onboarding-welcome' }"
            >
              Get started
            </KButton>
          </div>
        </div>
      </template>
    </KAlert>
  </div>
</template>

<script lang="ts" setup>
import { KAlert, KButton } from '@kong/kongponents'
import { ref } from 'vue'

import { useI18n } from '@/utilities'

const { t } = useI18n()
const alertClosed = ref(false)

function closeAlert() {
  alertClosed.value = true
}
</script>

<style lang="scss" scoped>
.onboarding-check {
  margin: 0 0 var(--spacing-xl) 0;
}

.alert-content {
  @media screen and (min-width: 700px) {
    display: flex;
    align-items: center;

    > *:first-of-type {
      margin-right: var(--spacing-md);
    }

    > *:last-of-type {
      min-width: 150px;
    }
  }

  @media screen and (max-width: 699px) {
    > *:last-of-type {
      margin-top: 10px;
    }
  }
}

.action-button.action-button {
  text-decoration: none;
}
</style>
