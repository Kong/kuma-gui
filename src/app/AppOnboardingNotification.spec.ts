import { mount } from '@vue/test-utils'

import AppOnboardingNotification from './AppOnboardingNotification.vue'

describe('AppOnboardingNotification', () => {
  it('renders snapshot', () => {
    const wrapper = mount(AppOnboardingNotification)

    expect(wrapper.element).toMatchSnapshot()
  })
})
