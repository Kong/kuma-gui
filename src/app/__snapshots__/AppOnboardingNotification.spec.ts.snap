// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppOnboardingNotification renders snapshot 1`] = `
<div
  class="onboarding-check"
>
  <div
    class="success icon k-alert dismissible"
    data-v-1910ef98=""
    role="alert"
  >
    <!---->
    <!---->
    <div
      class="k-alert-msg-text has-dismiss-icon"
      data-v-1910ef98=""
    >
      <!---->
      <div
        class="k-alert-msg"
        data-v-1910ef98=""
      >
        
        <div
          class="alert-content"
        >
          <div>
            <strong>
              Welcome to Kuma!
            </strong>
             We've detected that you don't have any data plane proxies running yet. We've created an onboarding process to help you! 
          </div>
          <div>
            <router-link
              class="small rounded primary k-button action-button"
              data-v-6f773baa=""
              to="[object Object]"
              type="button"
            >
              
              <!---->
              
              
               Get started 
              
              <!---->
            </router-link>
          </div>
        </div>
        
      </div>
      <!---->
    </div>
    <button
      aria-label="Close"
      class="close"
      data-v-1910ef98=""
      type="button"
    >
      <span
        class="success kong-icon-close kong-icon"
        data-v-1910ef98=""
        data-v-71cfcfc4=""
      >
        <svg
          fill="none"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          
  
          <title>
            Close
          </title>
          
  
          <path
            d="M16 4 3 17m13 0L3 4"
            stroke="#A3BBCC"
            stroke-linecap="round"
            stroke-width="2"
          />
          

        </svg>
        

      </span>
    </button>
    <div
      class="success k-alert-action"
      data-v-1910ef98=""
    >
      <!---->
      <!---->
    </div>
  </div>
</div>
`;
